consensus/src/consensus.rs

```diff
@@ -191,11 +183,14 @@ impl<S: Storage> Consensus<S> {
                         side_chain_path.new_block_number
                     );
 
-                    // If the side chain is now longer than the canon chain,
+                    // If the side chain is now heavier than the canon chain,
                     // perform a fork to the side chain.
-                    if side_chain_path.new_block_number > self.ledger.get_current_block_height() {
+                    let canon_difficulty =
+                        self.get_canon_difficulty_from_height(side_chain_path.shared_block_number)?;
+
+                    if side_chain_path.aggregate_difficulty > canon_difficulty {
                         debug!(
-                            "Determined side chain is longer than canon chain by {} blocks",
+                            "Determined side chain is heavier than canon chain by {} blocks",
```

Comment:
nit: some percentage or other value might be more informative than the height difference now

Reply:
Good point :) I'll adjust it!